src/jdk.jlink/share/classes/jdk/tools/jlink/internal/JlinkTask.java

```diff
@@ -569,24 +588,28 @@ public Set<ModuleReference> findAll() {
      * version or the java.base version is not the same as the current runtime's
      * version.
      */
-    private static void checkJavaBaseVersion(ModuleFinder finder) {
-        assert finder.find("java.base").isPresent();
-
-        // use the version of java.base module, if present, as
-        // the release version for multi-release JAR files
-        ModuleDescriptor.Version v = finder.find("java.base").get()
-                .descriptor().version().orElseThrow(() ->
-                new IllegalArgumentException("No version in java.base descriptor")
-                        );
-
-        Runtime.Version version = Runtime.Version.parse(v.toString());
-        if (Runtime.version().feature() != version.feature() ||
-                Runtime.version().interim() != version.interim()) {
-            // jlink version and java.base version do not match.
-            // We do not (yet) support this mode.
-            throw new IllegalArgumentException(taskHelper.getMessage("err.jlink.version.mismatch",
-                    Runtime.version().feature(), Runtime.version().interim(),
-                    version.feature(), version.interim()));
+    private void checkJavaBaseVersion(ModuleFinder finder) {
+        try {
+            ModuleReference current = ModuleLayer.boot()
+                    .configuration()
+                    .findModule("java.base")
+                    .get()
+                    .reference();
+            ModuleReference target = finder.find("java.base").get();
+            String currentRelease = getReleaseInfo(current).orElseThrow(() ->
+                new IllegalArgumentException("Cannot find release.txt"));
+            String targetRelease = getReleaseInfo(target).orElse("N/A");
+            if (! currentRelease.equals(targetRelease)) {
+                // jlink version and java.base version do not match.
```

Comment:
The comment could be clearer. It's a version mismatch between the current runtime and the target runtime image.